feat: FC-0012 - add (--no-segment) option to extract_translations

shadinaif opened the pull request:
<!--

🌴🌴
🌴🌴🌴🌴         🌴 Note: Palm is in support. Fixes you make on master may still be needed on Palm.
    🌴🌴🌴🌴     If so, make another pull request against the open-release/palm.master branch,
🌴🌴🌴🌴         or ask in the #wg-build-test-release Slack channel if you have any questions or need help.
🌴🌴

Please give your pull request a short but descriptive title.
Use conventional commits to separate and summarize commits logically:
https://open-edx-proposals.readthedocs.io/en/latest/oep-0051-bp-conventional-commits.html

Use this template as a guide. Omit sections that don't apply.
You may link to information rather than copy it, but only if the link is publicly
readable.  If you must linked information must be private (because it has secrets),
clearly label the link as private.

-->

## Description

Add `--no-segment` option to `extract_translations`. The flag is added if and only if `IS_OPENEDX_TRANSLATIONS_WORKFLOW` is set to `yes`. This a follow-up work to [Decision to standardize django/djangojs po files](https://github.com/openedx/edx-platform/pull/32994)

`IS_OPENEDX_TRANSLATIONS_WORKFLOW` flag is introduced and used by [openedx-translations](https://github.com/openedx/openedx-translations)

- [x] Tested with and without the flag. Working fine
- [x] Tested with openedx-translation fork https://github.com/Zeit-Labs/openedx-translations/pull/89/files#diff-dca191d363b47b36a676551d004f32d72595eb6bd57096aff82794c40e6d4c87

References
----------

This contribution is part of the [FC-0012 project](https://openedx.atlassian.net/l/cp/XGS0iCcQ) which is sparked by the [Translation Infrastructure update OEP-58](https://open-edx-proposals.readthedocs.io/en/latest/architectural-decisions/oep-0058-arch-translations-management.html#specification).

Up-to-date project overview and details are available in the [Approach Memo and Technical Discovery: Translations Infrastructure Implementation](https://docs.google.com/document/d/11dFBCnbdHiCEdZp3pZeHdeH8m7Glla-XbIin7cnIOzU/edit#) document.

Join the conversation on [Open edX Slack #translations-project-fc-0012](https://openedx.slack.com/archives/C04R6TUJB7T).

Check the links above for full information about the overall project.

Internalization is being re-architected in Open edX Python, XBlock, Micro-frontend, and other projects. There are a number of immediately visible changes:
 - Remove source and language translations from the repositories, hence no `.json`, `.po` or `.mo` files will be committed into the repos.
 - Add standardized `make extract_translations` in all repositories
 - Push user-facing messages strings into [openedx/openedx-translations](https://github.com/openedx/openedx-translations/).
 - Integrate root repositories with [openedx/openedx-atlas](https://github.com/openedx/openedx-atlas/) to pull translations on build/deploy time

Breaking Changes
----------------

One of the primary goals of the project is to **avoid breaking changes**. If you notice any suspicious code, please raise your concern. But before that, please know the strategy we're following to avoid breaking changes

openedx-webhooks commented:
Thanks for the pull request, @shadinaif! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.




<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

OmarIthawi commented on Makefile:
```diff
@@ -38,7 +38,17 @@ guides:	swagger ## build the developer guide docs
-	i18n_tool extract -v
+	# (IS_OPENEDX_TRANSLATIONS_WORKFLOW) is set to "yes" when openedx-translations calls this target.
+	# Otherwise, we preserve the old behavior of having segmented po files.
+	if [ "$(IS_OPENEDX_TRANSLATIONS_WORKFLOW)" = "yes" ]; then \
```
I recommend using `ifeq` Makefile to ensure it's compatible across operating systems: [example](https://github.com/openedx/frontend-app-gradebook/blob/1c704585908d21aa4ccee40affb29a086f29067a/Makefile#L58-L75).

It also makes the `Makefile` more readable because less `/` slashes are appended to the script.

shadinaif replied:
done

OmarIthawi commented on Makefile:
```diff
@@ -38,7 +38,17 @@ guides:	swagger ## build the developer guide docs
+		cd conf/locale/en/LC_MESSAGES; \
+		msgcat django.po wiki.po edx_proctoring_proctortrack.po mako.po underscore.po -o django_temp.po; \
+		mv django_temp.po django.po; \
+		rm wiki.po edx_proctoring_proctortrack.po mako.po underscore.po; \
```
`underscore` should be merged into `djangojs.po`:

For demonstration purposes:

```
msgcat django.po mako.po wiki.po edx_proctoring_proctortrack.po -o django.po
msgcat djangojs.po underscore.po -o djangojs.po
```

shadinaif replied:
done

OmarIthawi reviewed (commented):
Thanks @shadinaif. I realize this is still work-in progress but I'm adding notes if it's useful.

OmarIthawi commented:
Thanks @shadinaif. Some tests are failing. 

Sometimes we only need to rebase over `master` and that's it.

OmarIthawi commented on Makefile:
```diff
@@ -37,8 +37,19 @@ technical-docs:  ## build the technical docs
+	cd conf/locale/en/LC_MESSAGES && msgcat djangojs.po underscore.po -o djangojs.po
+	cd conf/locale/en/LC_MESSAGES && msgcat django.po wiki.po edx_proctoring_proctortrack.po mako.po -o django_temp.po
+	cd conf/locale/en/LC_MESSAGES && mv django_temp.po django.po
+	cd conf/locale/en/LC_MESSAGES && rm wiki.po edx_proctoring_proctortrack.po mako.po underscore.po
```
```suggestion
	cd conf/locale/en/LC_MESSAGES && msgcat djangojs.po underscore.po -o djangojs.po
	cd conf/locale/en/LC_MESSAGES && msgcat django.po wiki.po edx_proctoring_proctortrack.po mako.po -o django.po
	cd conf/locale/en/LC_MESSAGES && rm wiki.po edx_proctoring_proctortrack.po mako.po underscore.po
```

shadinaif replied:
done

OmarIthawi reviewed (approved):
Thanks @shadinaif! 

This looks great!

Would you mind checking my last note? Same code but one less line.

OmarIthawi commented on Makefile:
```diff
@@ -37,8 +37,19 @@ technical-docs:  ## build the technical docs
 	cd docs/guides; make clean html
 
+# (IS_OPENEDX_TRANSLATIONS_WORKFLOW) is set to "yes" when openedx-translations calls this target.
+# Otherwise, we preserve the old behavior of having segmented po files.
```
```suggestion
# (IS_OPENEDX_TRANSLATIONS_WORKFLOW) is set to "yes" when openedx-translations repository extracts translations. 
# Related doc: https://docs.openedx.org/en/latest/developers/how-tos/enable-translations-new-repo.html
# Otherwise, we preserve the old behavior of having segmented po files.
```

shadinaif replied:
done

OmarIthawi commented:
@shadinaif I've added two minor notes. Please test it on the `openedx-translations` repo to ensure the workflow is working.

shadinaif commented:
Tested, and ready to be merged @OmarIthawi @brian-smith-tcril

OmarIthawi commented:
@jmbowman this is another FC-0012 pull request that needs your review.

OmarIthawi commented:
@shadinaif please add the `FC-0012` suffix to the title.

brian-smith-tcril commented on Makefile:
```diff
@@ -37,8 +37,19 @@ technical-docs:  ## build the technical docs
 guides:	swagger ## build the developer guide docs
 	cd docs/guides; make clean html
 
+# (IS_OPENEDX_TRANSLATIONS_WORKFLOW) is set to "yes" when openedx-translations repository extracts translations.
```
```suggestion
# (IS_OPENEDX_TRANSLATIONS_WORKFLOW) is set to "yes" in the `extract-translation-source-files` GitHub actions workflow on the `openedx-translations` repository
```

might even be worth it to throw a link to the repo/workflow file in here

shadinaif replied:
done

brian-smith-tcril commented on Makefile:
```diff
@@ -37,8 +37,19 @@ technical-docs:  ## build the technical docs
 	cd docs/guides; make clean html
 
+# (IS_OPENEDX_TRANSLATIONS_WORKFLOW) is set to "yes" when openedx-translations repository extracts translations.
+# Related doc: https://docs.openedx.org/en/latest/developers/how-tos/enable-translations-new-repo.html
```
it's not clear reading through this doc how it is related to this. I think it makes more sense to link to https://github.com/openedx/edx-platform/blob/59ddee7ad3363bf9f4bfae355fa3f24a1bbdfb37/docs/decisions/0018-standarize-django-po-files.rst once it is merged instead.

brian-smith-tcril commented on Makefile:
```diff
@@ -37,8 +37,19 @@ technical-docs:  ## build the technical docs
 
+# (IS_OPENEDX_TRANSLATIONS_WORKFLOW) is set to "yes" when openedx-translations repository extracts translations.
+# Related doc: https://docs.openedx.org/en/latest/developers/how-tos/enable-translations-new-repo.html
+# Otherwise, we preserve the old behavior of having segmented po files.
```
```suggestion
```

OmarIthawi commented:
@dianakhuang would you mind taking a look on this pull request?

@shadinaif please let me know if there's anything we can do to push the PR forward.

OmarIthawi commented:
@shadinaif would you mind checking the notes in the review?

shadinaif committed:
feat: add (--no-segment) option to extract_translations

The flag is added if and only if (IS_OPENEDX_TRANSLATIONS_WORKFLOW) is set to "yes"

Refs: FC-0012 OEP-58

shadinaif commented:
Tested [here](https://github.com/Zeit-Labs/openedx-translations/pull/87/files) @OmarIthawi , please review

OmarIthawi reviewed (approved):
Looks great! Thanks @shadinaif!

brian-smith-tcril commented:
@dianakhuang you're listed as "Review required from" for this repo on the spreadsheet. Any chance you could review/merge this one?

dianakhuang reviewed (approved):
yup @brian-smith-tcril ! It was on my radar, but I figured I'd let everyone else get to it first. Looks good to me.

openedx-webhooks commented:
@shadinaif 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-platform&entry.867055334=https://github.com/openedx/edx-platform/pull/33368&entry.1484655318=https://github.com/shadinaif&entry.752974735=2023-09-28+06:18&entry.1917517419=2023-11-15+17:28&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
